### PR TITLE
A kruglov

### PR DESCRIPTION
**Pull request by Darshniderd**
Add two task (3-4) from PyLec4

### CODE REVIEW

File: Practice/kruglov/home_task_4/PyLec4T3-4.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# 3
```
Comment (IlyaOrlov) [inline Practice/kruglov/home_task_4/PyLec4T3-4.py]: Good solution, but the task expects using of a single list.

File: Practice/kruglov/home_task_4/PyLec4T3-4.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+s = len(d)
+while i < s:
+    c = min(d)
+    b.append(c)
+    d.pop(d.index(c))
+    i += 1
+
+print(b)
+
+# 3.1
```
Comment (IlyaOrlov) [inline Practice/kruglov/home_task_4/PyLec4T3-4.py]: Almost meets the task) But it's expected that u-element will take the place of min element as well.

File: Practice/kruglov/home_task_4/PyLec4T3-4.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+g = [0, 3, 24, 2, 3, 7]
+
+u = 0
+while u < len(g):
+    h = g.pop(g.index(min(g[u:]), u))
+    g.insert(u, h)
+    u += 1
+print(g)
+
+# 4
```
Comment (IlyaOrlov) [inline Practice/kruglov/home_task_4/PyLec4T3-4.py]: OK

File: Practice/kruglov/home_task_4/PyLec4T3-4.py (hunk 1 above)
Comment (Darshniderd) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: Noted how - "Poor solution"

File: Practice/kruglov/home_task_4/PyLec4T3-4.py (hunk 2 above)
Comment (Darshniderd) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: Fixed

File: Practice/kruglov/home_task_4/PyLec4T3-4.py (hunk 3 above)
Comment (Darshniderd) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: )

File: Practice/kruglov/home_task_4/PyLec4T3-4.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# print(b)
+
+# 3.1
+list1 = [0, 3, 24, 2, 3, 7]
+el = 0
+len_list = len(list1)
+while el < len_list:
+    min_ind = list1.index(min(list1[el:]), el)
+    min_val = list1.pop(min_ind)
```
Comment (IlyaOrlov) [inline Practice/kruglov/home_task_4/PyLec4T3-4.py]: For lines 21-25: there's more pythonic way to perform this swap: 
```python
list1[min_ind], list1[el] = list1[el], list1[min_ind]
```

Comment (Darshniderd) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: This is some kind of magic!

Comment (Darshniderd) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: Fixed

Comment (IlyaOrlov) [reply Practice/kruglov/home_task_4/PyLec4T3-4.py]: It's not magic, it's Python:)

**Pull request merged**